### ISSUE TITLE
Cash transaction missing fields

### ISSUE THREAD

**Issue by joshxg** [openmsupply/mobile#2379]
## Is your feature request related to a problem? Please describe.

There are a few missing fields on a cash transaction

## Describe the solution you'd like

cc for a `CashIn`:
- `entry_time`
- `paymentType`

CI for a `CashOut`
- `linked_transaction_id` : This is set to point to the PS, but should be an empty string.
- `PaymentTypeID` : not set.

PS For a `CashOut`:
- `entry_time`
- `option_ID` not set
- the `PS` trans_line `foreign_currency_price` is set - but in desktop it isn't.
- the `PS` trans_line `line_number` is 0, should start from 1.

## Implementation

**A description of how the solution would best be implemented in the code base**

## Describe alternatives you've considered

**A clear and concise description of any alternative solutions or features you've considered.**

**Comment by joshxg** [comment]
```
[
  {
    "name_ID": "373C5553052C43ED950BCF4C03FC0824",
    "ID": "C18B5E6822904178817934817A48E836",
    "invoice_num": 25,
    "amount_outstanding": 0,
    "comment": "Receipt for invoice(s) 28",
    "entry_date": "2020-02-24T11:00:00.000Z",
    "type": "rc",
    "status": "fn",
    "total": 5,
    "export_batch": 0,
    "linked_transaction_id": "",
    "their_ref": "",
    "confirm_date": "2020-02-24T11:00:00.000Z",
    "service_descrip": "",
    "service_price": 0,
    "subtotal": 0,
    "tax": 0,
    "user_ID": "1A7D48B4FC366C40B530F3E449A2C150",
    "pickslip_printed_date": "0000-00-00T00:00:00.000Z",
    "prescriber_ID": "",
    "goods_received_ID": "",
    "invoice_printed_date": "0000-00-00T00:00:00.000Z",
    "ship_date": "0000-00-00T00:00:00.000Z",
    "ship_method_ID": "",
    "ship_method_comment": "",
    "waybill_number": "",
    "number_of_cartons": 0,
    "arrival_date_estimated": "0000-00-00T00:00:00.000Z",
    "arrival_date_actual": "0000-00-00T00:00:00.000Z",
    "responsible_officer_ID": "",
    "mode": "store",
    "category_ID": "",
    "confirm_time": 49806,
    "foreign_currency_total": 0,
    "currency_ID": "A5F808D9977DB04D8698D950AE403117",
    "hold": false,
    "currency_rate": 0,
    "supplier_charge_fc": 0,
    "local_charge_distributed": 0,
    "budget_period_ID": "",
    "store_ID": "121094487F327843B7B3D0ABB845C220",
    "user1": "",
    "user2": "",
    "mwks_sequence_num": 0,
    "is_cancellation": false,
    "user3": "",
    "user4": "",
    "Colour": 0,
    "original_PO_ID": "",
    "donor_default_id": "",
    "Date_order_received": "0000-00-00T00:00:00.000Z",
    "Date_order_written": "0000-00-00T00:00:00.000Z",
    "contact_id": "",
    "encounter_id": "",
    "is_authorised": false,
    "requisition_ID": "",
    "entry_time": 49806,
    "linked_goods_received_ID": "",
    "authorisationStatus": "",
    "nameInsuranceJoinID": "",
    "insuranceDiscountAmount": 0,
    "optionID": "",
    "insuranceDiscountRate": 0,
    "internalData": null,
    "lastModifiedAt": 1582638607,
    "custom_data": null,
    "goodsReceivedConfirmation": null,
    "paymentTypeID": "33B5E1F46D4FFD419951B29EE34F0B35",
    "Link_247": { "__KEY": "1A7D48B4FC366C40B530F3E449A2C150" },
    "transaction_category": { "__KEY": "" },
    "Link_140": { "__KEY": "" },
    "Link_156": { "__KEY": "" },
    "clinician": { "__KEY": "" },
    "name": { "__KEY": "373C5553052C43ED950BCF4C03FC0824" },
    "Link_226": { "__KEY": "" },
    "store": { "__KEY": "121094487F327843B7B3D0ABB845C220" },
    "Link_130": { "__KEY": "" },
    "nameInsuranceJoin": { "__KEY": "" },
    "requisition": { "__KEY": "" },
    "options": { "__KEY": "" },
    "transLines": [
      {
        "transaction_ID": "C18B5E6822904178817934817A48E836",
        "item_ID": "",
        "batch": "",
        "price_extension": 5,
        "note": "",
        "sell_price": 0,
        "expiry_date": "0000-00-00T00:00:00.000Z",
        "cost_price": 0,
        "pack_size": 0,
        "quantity": 0,
        "box_number": "",
        "item_line_ID": "",
        "line_number": 0,
        "item_name": "",
        "ID": "3AFD1C1E8B5F4F03A39392E4EE25B6FD",
        "supp_trans_line_ID_ns": "",
        "goods_received_lines_ID": "",
        "manufacturer_ID": "",
        "foreign_currency_price": 0,
        "location_ID": "",
        "volume_per_pack": 0,
        "repeat_ID": "",
        "user_1": "",
        "user_2": "",
        "user_3": "",
        "user_4": "",
        "pack_size_inner": 0,
        "pack_inners_in_outer": 0,
        "is_from_inventory_adjustment": false,
        "Weight": 0,
        "source_backorder_id": "",
        "order_lines_ID": "",
        "donor_id": "",
        "local_charge_line_total": 0,
        "type": "cash_in",
        "linked_transact_id": "A6D716CD484741948F225181119111FB",
        "user_5_ID": "",
        "user_6_ID": "",
        "user_7_ID": "",
        "user_8_ID": "",
        "linked_trans_line_ID": "",
        "barcodeID": "",
        "sentQuantity": 0,
        "optionID": "",
        "isVVMPassed": "",
        "doses": 0,
        "prescribedQuantity": 0,
        "item": { "__KEY": "" },
        "Link_225": { "__KEY": "" },
        "itemLine": { "__KEY": "" },
        "transact": { "__KEY": "C18B5E6822904178817934817A48E836" },
        "Link_258": { "__KEY": "" },
        "Link_259": { "__KEY": "" },
        "Link_260": { "__KEY": "" },
        "Link_261": { "__KEY": "" },
        "Link_272": { "__KEY": "" },
        "options": { "__KEY": "" },
        "linkedTransact": { "__KEY": "A6D716CD484741948F225181119111FB" }
      }
    ]
  },
  {
    "name_ID": "373C5553052C43ED950BCF4C03FC0824",
    "ID": "A6D716CD484741948F225181119111FB",
    "invoice_num": 28,
    "amount_outstanding": 0,
    "comment": "",
    "entry_date": "2020-02-24T11:00:00.000Z",
    "type": "ci",
    "status": "fn",
    "total": 5,
    "export_batch": 0,
    "linked_transaction_id": "",
    "their_ref": "",
    "confirm_date": "2020-02-24T11:00:00.000Z",
    "service_descrip": "",
    "service_price": 0,
    "subtotal": 5,
    "tax": 0,
    "user_ID": "1A7D48B4FC366C40B530F3E449A2C150",
    "pickslip_printed_date": "0000-00-00T00:00:00.000Z",
    "prescriber_ID": "",
    "goods_received_ID": "",
    "invoice_printed_date": "0000-00-00T00:00:00.000Z",
    "ship_date": "0000-00-00T00:00:00.000Z",
    "ship_method_ID": "",
    "ship_method_comment": "",
    "waybill_number": "",
    "number_of_cartons": 0,
    "arrival_date_estimated": "0000-00-00T00:00:00.000Z",
    "arrival_date_actual": "0000-00-00T00:00:00.000Z",
    "responsible_officer_ID": "",
    "mode": "dispensary",
    "category_ID": "",
    "confirm_time": 49806,
    "foreign_currency_total": 5,
    "currency_ID": "A5F808D9977DB04D8698D950AE403117",
    "hold": false,
    "currency_rate": 1,
    "supplier_charge_fc": 0,
    "local_charge_distributed": 0,
    "budget_period_ID": "",
    "store_ID": "121094487F327843B7B3D0ABB845C220",
    "user1": "inpatient",
    "user2": "",
    "mwks_sequence_num": 0,
    "is_cancellation": false,
    "user3": "",
    "user4": "",
    "Colour": 1710361,
    "original_PO_ID": "",
    "donor_default_id": "",
    "Date_order_received": "0000-00-00T00:00:00.000Z",
    "Date_order_written": "2020-02-24T11:00:00.000Z",
    "contact_id": "",
    "encounter_id": "",
    "is_authorised": false,
    "requisition_ID": "",
    "entry_time": 49246,
    "linked_goods_received_ID": "",
    "authorisationStatus": "",
    "nameInsuranceJoinID": "",
    "insuranceDiscountAmount": 0,
    "optionID": "",
    "insuranceDiscountRate": 0,
    "internalData": null,
    "lastModifiedAt": 1582638607,
    "custom_data": null,
    "goodsReceivedConfirmation": null,
    "paymentTypeID": "",
    "Link_247": { "__KEY": "1A7D48B4FC366C40B530F3E449A2C150" },
    "transaction_category": { "__KEY": "" },
    "Link_140": { "__KEY": "" },
    "Link_156": { "__KEY": "" },
    "clinician": { "__KEY": "" },
    "name": { "__KEY": "373C5553052C43ED950BCF4C03FC0824" },
    "Link_226": { "__KEY": "" },
    "store": { "__KEY": "121094487F327843B7B3D0ABB845C220" },
    "Link_130": { "__KEY": "" },
    "nameInsuranceJoin": { "__KEY": "" },
    "requisition": { "__KEY": "" },
    "options": { "__KEY": "" },
    "transLines": [
      {
        "transaction_ID": "A6D716CD484741948F225181119111FB",
        "item_ID": "769CD2C89F85EC488D4E387484F9D050",
        "batch": "8076484",
        "price_extension": 5,
        "note": "",
        "sell_price": 5,
        "expiry_date": "2020-05-30T12:00:00.000Z",
        "cost_price": 0,
        "pack_size": 1,
        "quantity": 1,
        "box_number": "",
        "item_line_ID": "C8FDBAEAE42E8045BE5A1D6DFD95F90A",
        "line_number": 2,
        "item_name": "ABACAVIR/LAMIVUDINE 600/300 MG CP",
        "ID": "C9E2B53159A2457793F88C5A3B9E594D",
        "supp_trans_line_ID_ns": "",
        "goods_received_lines_ID": "",
        "manufacturer_ID": "",
        "foreign_currency_price": 5,
        "location_ID": "80BC39BF3C91BB44896CD954E1EAC35F",
        "volume_per_pack": 0,
        "repeat_ID": "",
        "user_1": "",
        "user_2": "",
        "user_3": "",
        "user_4": "",
        "pack_size_inner": 0,
        "pack_inners_in_outer": 0,
        "is_from_inventory_adjustment": false,
        "Weight": 0,
        "source_backorder_id": "",
        "order_lines_ID": "",
        "donor_id": "",
        "local_charge_line_total": 0,
        "type": "stock_out",
        "linked_transact_id": "",
        "user_5_ID": "",
        "user_6_ID": "",
        "user_7_ID": "",
        "user_8_ID": "",
        "linked_trans_line_ID": "",
        "barcodeID": "",
        "sentQuantity": 0,
        "optionID": "",
        "isVVMPassed": "",
        "doses": 0,
        "prescribedQuantity": 0,
        "item": { "__KEY": "769CD2C89F85EC488D4E387484F9D050" },
        "Link_225": { "__KEY": "" },
        "itemLine": { "__KEY": "C8FDBAEAE42E8045BE5A1D6DFD95F90A" },
        "transact": { "__KEY": "A6D716CD484741948F225181119111FB" },
        "Link_258": { "__KEY": "" },
        "Link_259": { "__KEY": "" },
        "Link_260": { "__KEY": "" },
        "Link_261": { "__KEY": "" },
        "Link_272": { "__KEY": "" },
        "options": { "__KEY": "" },
        "linkedTransact": { "__KEY": "" }
      },
      {
        "transaction_ID": "A6D716CD484741948F225181119111FB",
        "item_ID": "D83832AB545BDC49BFBECF50B5C98AAF",
        "batch": "PA70515",
        "price_extension": 0,
        "note": "",
        "sell_price": 0,
        "expiry_date": "2020-03-30T11:00:00.000Z",
        "cost_price": 0,
        "pack_size": 1,
        "quantity": 5,
        "box_number": "",
        "item_line_ID": "9E4F844EA600614094C520DB0FA1814E",
        "line_number": 1,
        "item_name": "ABACAVIR / LAMIVUDINE 60/30 MG CP",
        "ID": "F913DB1E935D472487F568BA95E4D5F0",
        "supp_trans_line_ID_ns": "",
        "goods_received_lines_ID": "",
        "manufacturer_ID": "",
        "foreign_currency_price": 0,
        "location_ID": "48666DC560A02C4FA2B58B9C8EE400AA",
        "volume_per_pack": 0,
        "repeat_ID": "",
        "user_1": "",
        "user_2": "",
        "user_3": "",
        "user_4": "",
        "pack_size_inner": 0,
        "pack_inners_in_outer": 0,
        "is_from_inventory_adjustment": false,
        "Weight": 0,
        "source_backorder_id": "",
        "order_lines_ID": "",
        "donor_id": "",
        "local_charge_line_total": 0,
        "type": "stock_out",
        "linked_transact_id": "",
        "user_5_ID": "",
        "user_6_ID": "",
        "user_7_ID": "",
        "user_8_ID": "",
        "linked_trans_line_ID": "",
        "barcodeID": "",
        "sentQuantity": 0,
        "optionID": "",
        "isVVMPassed": "",
        "doses": 0,
        "prescribedQuantity": 0,
        "item": { "__KEY": "D83832AB545BDC49BFBECF50B5C98AAF" },
        "Link_225": { "__KEY": "" },
        "itemLine": { "__KEY": "9E4F844EA600614094C520DB0FA1814E" },
        "transact": { "__KEY": "A6D716CD484741948F225181119111FB" },
        "Link_258": { "__KEY": "" },
        "Link_259": { "__KEY": "" },
        "Link_260": { "__KEY": "" },
        "Link_261": { "__KEY": "" },
        "Link_272": { "__KEY": "" },
        "options": { "__KEY": "" },
        "linkedTransact": { "__KEY": "" }
      }
    ]
  }
]
```

**Comment by joshxg** [comment]
Closed by #2399

**Comment by wlthomson** [comment]
Tested on `v5.0.0-rc3`:

- [x] The offset customer credit created from a cash in transaction has the payment type of the cash in transaction when synced to desktop
- [x] The offset customer credit created from a cash in transaction has no linked transaction when synced to desktop
- [x] The offset customer invoice created from a cash out transaction has the payment type of the cash out transaction when synced to desktop
- [x] The offset customer invoice created from a cash out transaction has no linked transaction out when synced to desktop
- [x] The offset customer invoice created from a cash out transaction has the reason type of the cash out transaction when synced to desktop
